src/main/java/com/sonymobile/tools/gerrit/gerritevents/ssh/SshConnectionImpl.java

```diff
@@ -400,6 +400,7 @@ public synchronized void disconnect() {
             logger.debug("Disconnecting client connection.");
             connectSession.disconnect();
             connectSession = null;
+            authentication.deleteTempKeyFile();
```

Comment:
Isn't it a bit dangerous to have the key lying around on the file system for that long? Especially in a temp folder?

Reply:
Agreed - it was designed to provide compatibility on with the existing behavior(read a key from disk). The key could be written to user home with limited readability. There's also no reason it has to stay this way per your other comment.